File: backend/src/main/java/br/ufpr/sept/so2/modules/iam/infrastructure/persistence/JtiBlacklistJpaAdapter.java
package br.ufpr.sept.so2.modules.iam.infrastructure.persistence;

import br.ufpr.sept.so2.modules.iam.application.ports.JtiBlacklistRepository;
import org.springframework.stereotype.Component;

import java.time.OffsetDateTime;

@Component
public class JtiBlacklistJpaAdapter implements JtiBlacklistRepository {

    private final JtiBlacklistJpaRepository jpaRepository;

    public JtiBlacklistJpaAdapter(JtiBlacklistJpaRepository jpaRepository) {
        this.jpaRepository = jpaRepository;
    }

    @Override
    public void add(String jti, OffsetDateTime expiresAt) {
        jpaRepository.save(new JtiBlacklistJpaEntity(jti, expiresAt, OffsetDateTime.now()));
    }

    @Override
    public boolean contains(String jti) {
        return jpaRepository.existsById(jti);
    }
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/iam/infrastructure/persistence/JtiBlacklistJpaAdapter.kt
package br.ufpr.sept.so2.modules.iam.infrastructure.persistence

import br.ufpr.sept.so2.modules.iam.application.ports.JtiBlacklistRepository
import org.springframework.stereotype.Component
import java.time.OffsetDateTime

@Component
class JtiBlacklistJpaAdapter(
    private val jpaRepository: JtiBlacklistJpaRepository,
) : JtiBlacklistRepository {
    override fun add(jti: String, expiresAt: OffsetDateTime) {
        jpaRepository.save(JtiBlacklistJpaEntity(jti, expiresAt, OffsetDateTime.now()))
    }

    override fun contains(jti: String): Boolean = jpaRepository.existsById(jti)
}
